### PR TITLE
[READY]: Issue-192 - AP config type switching

### LINKED ISSUE #192: Configure AP to consume ap-network-type parameter
- Create script `/root/bin/config-version.sh` to symlink `/etc/config/network` and `/etc/config/wireless` to type specific versions of the file in `network.$number` or `wireless.$number` format
- `config-version.sh` should take an integer as argument, assuming `0` if omitted, which verifies proper symlinking. always defaulting to `0` if state is uncertain
- Add `config-version.sh` to udhcp.user hook script and call based on which version is passed via dhcp
- Add initial call to `config-version.sh` to `rc.local`

### PR DESCRIPTION
**Pull request by kylerisse**
## Description of PR
Fixes #192 
Adds AP config types. Allows DHCP to update which version of configurations are running on an AP.

## Previous Behavior
Single static config

## New Behavior
* Script sets config "version" using symlinks
* Script runs on boot setting wireless and network config to version 0
* Script runs when DHCP passes AP config type

Additional added by @sarcasticadmin:
* Move config to version 0
* Include symlinks during `make build-img`
* Add serverspec tests to check for symlinks

## Tests
NOC, using spare AP & Robs test AP (wndr3800ch)

Additionally @sarcasticadmin:
* Built img on test AP (3800ch)
* `serverspec` passes on the same AP:

```
$ rake spec TEST_TYPE=openwrt TARGET_HOST=192.168.254.100
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home
/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb
                             
shared                           
  Command "which apinger 2> /dev/null"
    exit_status      
      should eq 0            
  Command "which bash 2> /dev/null"
    exit_status             
      should eq 0    
  Command "which logrotate 2> /dev/null"
    exit_status                  
      should eq 0            
  Command "which python3 2> /dev/null"
    exit_status              
      should eq 0                
  Command "which rsyslogd 2> /dev/null"
    exit_status                                               
      should eq 0
  Command "which zabbix_agentd 2> /dev/null"
    exit_status
      should eq 0
  Command "pgrep apinger"
    exit_status
      should eq 0
  Command "pgrep crond"
    exit_status
      should eq 0
  Command "pgrep rsyslogd"
    exit_status
      should eq 0
  Command "pgrep zabbix"
    exit_status
      should eq 0
  Command "rsyslogd -N1"
    exit_status
      should eq 0
File "/root/bin/wifi-details.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  File "/root/bin/config-version.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  Command "/root/bin/config-version.sh"
    exit_status
      should eq 0
  Command "/root/bin/config-version.sh -c 9999"
    exit_status
      should eq 1
  File "/etc/resolv.conf"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/network"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/wireless"
    should be symlink
    should be owned by "root"
    should be grouped into "root"

Finished in 18.39 seconds (files took 0.45102 seconds to load)
30 examples, 0 failures
```

### CODE REVIEW

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+usage(){
+  cat << EOF
+usage: "$(basename "$0")" [OPTIONS] ARGS
```
Comment (davidelang) [inline tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: if there is no config version specified, it should return the config in use (and possibly the filename list??)

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      usage
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND-1))
+
+RELOAD=0
+
+for file in ${FILES[*]}
```
Comment (davidelang) [inline tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: This should be done in two passes, if _any_ of the files for the config don't exist, we don't want to switch any of them

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  # exit if the target version doesn't exist
+  if [ ! -f "$CONFROOT/$file.$CONFVER" ]
+  then
+    echo "no file $CONFROOT/$file.$CONFVER"
+    exit 1
+  fi
+  # create symlinks if the current version isn't active
+  if [ ! -L "$CONFROOT/$file" ] || [ ! "$(readlink $CONFROOT/$file)" = "$CONFROOT/$file.$CONFVER" ]
+  then
+    rm -f $CONFROOT/$file
```
Comment (davidelang) [inline tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: to be on the safe side, can we verify that this is a symlink before removing it?

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 1 above)
Comment (sarcasticadmin) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: :+1: I can see that as better behavior, I think given that the filename list is static it should just return the config version that's in use

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 2 above)
Comment (sarcasticadmin) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: Good idea, simple enough to add

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 3 above)
Comment (sarcasticadmin) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: definitely

**Comment by sarcasticadmin** [comment]
@davidelang thanks reviewing! Ill work on incorporating these suggestions over the weekend and let you know when its ready.

**Comment by davidelang** [comment]
On Fri, 2 Aug 2019, Robert James Hernandez wrote:

> I think given that the filename list is static it should just return the 
> config version that's in use

that's why I was questioning it, but the list could change from version to 
version over time, which is why I thought there may be a reason to report it.

David Lang

**Comment by sarcasticadmin** [comment]
@davidelang definitely a good point, thanks for clarifying

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 1 above)
Comment (owendelong) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: I think @sarcasticadmin has the right idea. Let's do that.

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 2 above)
Comment (owendelong) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: Agreed. Let's hold get that implemented before we merge this PR

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 3 above)
Comment (owendelong) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: Please incorporate that prior to merging this PR as well.

**Comment by sarcasticadmin** [comment]
@davidelang @owendelong @kylerisse here are the results of my last version of `config-version.sh` and I updated the serverspec tests to include a few more validations for `config-version.sh`:

outputs current version:
```                                                           
root@OpenWrt:~# config-version.sh                                          
0                                                               
root@OpenWrt:~# ls -lah /etc/config/                            
drwxr-xr-x    1 root     root           0 Sep 25 05:06 .    
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..                  
-rw-------    1 root     root         363 Sep 25 05:06 lldpd    
lrwxrwxrwx    1 root     root           9 Sep 25 05:06 network -> network.0
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0 
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd 
-rw-r--r--    1 root     root         604 Sep 25 05:06 system  
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv              
lrwxrwxrwx    1 root     root          10 Sep 25 05:06 wireless -> wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
```

Changing config versions (network and wireless):
```
root@OpenWrt:~# cd /etc/config/
root@OpenWrt:/etc/config# config-version.sh -c 0
no change to config symlinks required
root@OpenWrt:/etc/config# config-version.sh -c 1
no file /etc/config/wireless.1
root@OpenWrt:/etc/config# cp wireless.0 wireless.1
root@OpenWrt:/etc/config# config-version.sh -c 1
no file /etc/config/network.1
root@OpenWrt:/etc/config# cp network.0 network.1
root@OpenWrt:/etc/config# config-version.sh -c 1
root@OpenWrt:/etc/config# config-version.sh
1                                                                                                                                                                                                                              
root@OpenWrt:/etc/config# ls -alh
drwxr-xr-x    1 root     root           0 Sep 25 18:11 .
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..
-rw-------    1 root     root         363 Sep 25 05:06 lldpd
lrwxrwxrwx    1 root     root           9 Sep 25 18:11 network -> network.1
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0
-rw-r--r--    1 root     root        1.3K Sep 25 18:11 network.1
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd
-rw-r--r--    1 root     root         604 Sep 25 05:06 system
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv
lrwxrwxrwx    1 root     root          10 Sep 25 18:11 wireless -> wireless.1
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 18:10 wireless.1
root@OpenWrt:/etc/config# config-version.sh -c 0
root@OpenWrt:/etc/config# config-version.sh
0
```

What happens with configs are mismatched, missing, or not symlinks:
```
root@OpenWrt:/etc/config# rm wireless
root@OpenWrt:/etc/config# ln -s wireless.1 wireless
root@OpenWrt:/etc/config# ls -alh
drwxr-xr-x    1 root     root           0 Sep 25 18:13 .
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..
-rw-------    1 root     root         363 Sep 25 05:06 lldpd
lrwxrwxrwx    1 root     root           9 Sep 25 18:11 network -> network.0
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0
-rw-r--r--    1 root     root        1.3K Sep 25 18:11 network.1
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd
-rw-r--r--    1 root     root         604 Sep 25 05:06 system
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv
lrwxrwxrwx    1 root     root          10 Sep 25 18:13 wireless -> wireless.1
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 18:10 wireless.1
root@OpenWrt:/etc/config# config-version.sh
/etc/config/network.0 doesnt match other config versions: wireless.1
root@OpenWrt:/etc/config# config-version.sh -c 1
/etc/config/network.0 doesnt match other config versions: wireless.1
root@OpenWrt:/etc/config# rm wireless
root@OpenWrt:/etc/config# cp wireless.1 wireless
root@OpenWrt:/etc/config# ls -alh
drwxr-xr-x    1 root     root           0 Sep 25 18:14 .
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..
-rw-------    1 root     root         363 Sep 25 05:06 lldpd
lrwxrwxrwx    1 root     root           9 Sep 25 18:11 network -> network.0
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0
-rw-r--r--    1 root     root        1.3K Sep 25 18:11 network.1
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd
-rw-r--r--    1 root     root         604 Sep 25 05:06 system
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv
-rw-r--r--    1 root     root        1.2K Sep 25 18:14 wireless
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 18:10 wireless.1
root@OpenWrt:/etc/config# config-version.sh
No symlink exists for /etc/config/wireless
root@OpenWrt:/etc/config# echo $?
1
root@OpenWrt:/etc/config# rm wireless
root@OpenWrt:/etc/config# config-version.sh
No symlink exists for /etc/config/wireless
root@OpenWrt:/etc/config# ls -alh
drwxr-xr-x    1 root     root           0 Sep 25 18:15 .
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..
-rw-------    1 root     root         363 Sep 25 05:06 lldpd
lrwxrwxrwx    1 root     root           9 Sep 25 18:11 network -> network.0
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0
-rw-r--r--    1 root     root        1.3K Sep 25 18:11 network.1
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd
-rw-r--r--    1 root     root         604 Sep 25 05:06 system
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 18:10 wireless.1
root@OpenWrt:/etc/config# ln -s wireless.0 wireless
root@OpenWrt:/etc/config# config-version.sh
0
root@OpenWrt:/etc/config#
```

serverspec:
```
 $ rake spec TEST_TYPE=openwrt TARGET_HOST=192.168.254.100                                                                                                                                                       
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home
/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb                                                                                                                     
                                                                                                                                                                                                                                        
shared                                                                                                                                                                                                                                  
  Command "which apinger 2> /dev/null"                                                                                                                                                                                                  
    exit_status                                                                                                                                                                                                                         
Please unlock the card                                                                                                                                                                                                                  
                                                                                                                                                                                                                                        
Number: 0005 00003A54                                                                                                                                                                                                                   
Holder: Robert James Hernandez                                                                                                                                                                                                          
PIN:                                                                                                                                                                                                                                    
      should eq 0                                                                                                                                                                                                                       
  Command "which bash 2> /dev/null"                                                                                                                                                                                                     
    exit_status                                                                                                                                                                                                                         
      should eq 0                                                                                                                                                                                                                       
  Command "which logrotate 2> /dev/null"                                                                                                                                                                                                
    exit_status                                                                                                                                                                                                                         
      should eq 0                                                                                                                                                                                                                       
  Command "which python3 2> /dev/null"                                                                                                                                                                                                  
    exit_status                                                                                                                                                                                                                         
      should eq 0                                                                                                                                                                                                                       
  Command "which rsyslogd 2> /dev/null"                                                                                                                                                                                                 
    exit_status                                                                                                                                                                                                                         
      should eq 0                                                                                                                                                                                                                       
  Command "which zabbix_agentd 2> /dev/null"                                                                                                                                                                                            
    exit_status                                                                                                                                                                                                                         
      should eq 0                                                                                                                                                                                                                       
  Command "pgrep apinger"                                                                                                                                                                                                               
    exit_status                                                                                                                                                                                                                         
      should eq 0
  Command "pgrep crond"                                                     
    exit_status                                                                    
      should eq 0                                                                
  Command "pgrep rsyslogd"                                                  
    exit_status                                                                                     
      should eq 0                                                                                       
  Command "pgrep zabbix"                                                                   
    exit_status                                                                                         
      should eq 0                                                                                   
  Command "rsyslogd -N1"                                                                   
    exit_status                                                                                   
      should eq 0                                                                
  File "/root/bin/wifi-details.sh"                                                            
    should exist                                                                 
    should be mode 750                                                                              
    should be owned by "root"                                                                           
    should be grouped into "root"                                                                           
  File "/root/bin/config-version.sh"                                                                        
    should exist                                                                                              
    should be mode 750                                                              
    should be owned by "root"                                                                   
    should be grouped into "root"                                                                                                     
  Command "/root/bin/config-version.sh"                                                                                                 
    exit_status                                                                                  
      should eq 0                                                                                                                       
  Command "/root/bin/config-version.sh -c 9999"                                                                                       
    exit_status                                                                                     
      should eq 1                                                                                                       
  File "/etc/resolv.conf"                                                        
    should be symlink                                                                                       
    should be owned by "root"                                                                                     
    should be grouped into "root"                                                                                                     
  File "/etc/config/network"                                                                                                            
    should be symlink                                                                           
    should be owned by "root"                                                                                      
    should be grouped into "root"                                                                                         
  File "/etc/config/wireless"                                                                           
    should be symlink                                                       
    should be owned by "root"                                          
    should be grouped into "root"                                                                                           
                                                                                                
Finished in 18.39 seconds (files took 0.45102 seconds to load)                         
30 examples, 0 failures
```

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 2 above)
Comment (sarcasticadmin) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: @davidelang heres the check: https://github.com/socallinuxexpo/scale-network/pull/203/files#diff-009c980269baa7501260e6c3f1b9e298R51-R53

File: tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh (hunk 3 above)
Comment (sarcasticadmin) [reply tests/unit/openwrt/golden/ipq806x/root/bin/config-version.sh]: @davidelang @owendelong  heres the check: https://github.com/socallinuxexpo/scale-network/pull/203/files#diff-009c980269baa7501260e6c3f1b9e298R51-R53

**Comment by davidelang** [comment]
there should also be a test for the case where a network.2 exists but a wireless.2 doesn't and where a wireless.3 exists but a network.3 doesn't (or similar)

File: openwrt/files/etc/rc.local [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Put your custom commands here that should be executed once
 # the system init finished. By default this file does nothing.
 
+# Default config version to 0
+/root/bin/config-version.sh
```
Comment (kylerisse) [inline openwrt/files/etc/rc.local]: What should the new behavior be on boot? This was prior to the changes of:

`# If config ver is not passed then print out current version and exit
if [ "$CONFVER" == "" ]; then
   echo ${CONFLIST[0]} |  awk -F. '{print $NF}'
   exit 0
fi`

Comment (sarcasticadmin) [reply openwrt/files/etc/rc.local]: Ah good catch, Im inclined to think we can just drop this all together. Im already creating the symlink as part of the build process: https://github.com/socallinuxexpo/scale-network/pull/203/files#diff-b0a351f9d8b5e7a78a6c384f198b231cR88

**Comment by sarcasticadmin** [comment]
@davidelang do you mean a test in `serverspec` or a test from the output I posted?

For `serverspec`: I'd like to not have serverspec creating files during its test runs. So until we actual have a config `wireless.2` and `network.2`, and so on in the iamge I think we shouldnt worry about this.

For the test output I posted: Yes it will account for all of those cases. The logic just looks for whatever you config you'd like to enable and if that config pair (network and wireless) dont exist then it fails. That way we can have an arbitrary number on configs.

One thing I would like to do in a follow up issue is create [bats](https://github.com/bats-core/bats-core) tests for our shell scripts (`config-version.sh` in particular). Then we can really go through all the various cases and continue to ensure they work as expected.

**Comment by sarcasticadmin** [comment]
@kylerisse I fixed up your latest commit into your original commit to the PR since thats where the `/etc/rc.local` change originate. Additionally I updated the golden tests too so the openwrt tests would pass.  I just built an image and loaded it onto my `wndr3800ch` and it looks good! See the result below:

`/etc/rc.local` looks like we'd expect now, and the initial `*.0` configs are symlinked from the initial build process:
```
$ ssh root@192.168.254.100                                                                                                                                                                                     


BusyBox v1.30.0 () built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r0-1adee6b
 -----------------------------------------------------
root@OpenWrt:~# cat /etc/rc.local
# Put your custom commands here that should be executed once
# the system init finished. By default this file does nothing.

# TODO make this a true service
/root/bin/wifi-details.sh >/dev/null 2>/dev/null </dev/null &

exit 0
root@OpenWrt:~# ls -lah /etc/config/
drwxr-xr-x    1 root     root           0 Sep 25 05:06 .
drwxr-xr-x    1 root     root           0 Sep 25 05:07 ..
-rw-------    1 root     root         363 Sep 25 05:06 lldpd
lrwxrwxrwx    1 root     root           9 Sep 25 05:06 network -> network.0
-rw-r--r--    1 root     root        1.3K Sep 25 05:06 network.0
-rw-r--r--    1 root     root        2.5K Sep 25 05:06 snmpd
-rw-r--r--    1 root     root         604 Sep 25 05:06 system
-rw-r--r--    1 root     root         115 Sep 25 05:06 ubootenv
lrwxrwxrwx    1 root     root          10 Sep 25 05:06 wireless -> wireless.0
-rw-r--r--    1 root     root        1.2K Sep 25 05:06 wireless.0
```

serverspec results:

```
$ rake spec TEST_TYPE=openwrt TARGET_HOST=192.168.254.100                                                                                                                                              [49/1717]
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home
/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb
                             
shared                           
  Command "which apinger 2> /dev/null" 
    exit_status                                                 
      should eq 0
  Command "which bash 2> /dev/null"            
    exit_status
      should eq 0
  Command "which logrotate 2> /dev/null"
    exit_status      
      should eq 0            
  Command "which python3 2> /dev/null"
    exit_status             
      should eq 0    
  Command "which rsyslogd 2> /dev/null"
    exit_status                  
      should eq 0            
  Command "which zabbix_agentd 2> /dev/null"
    exit_status              
      should eq 0                
  Command "pgrep apinger"
    exit_status                                              
      should eq 0      
  Command "pgrep crond"
    exit_status                                   
      should eq 0
  Command "pgrep rsyslogd"
    exit_status                        
      should eq 0
  Command "pgrep zabbix"                        
    exit_status                                   
      should eq 0                                  
  Command "rsyslogd -N1"                           
    exit_status                               
      should eq 0                                     
  File "/root/bin/wifi-details.sh"
    should exist                                      
    should be mode 750           
    should be owned by "root"                               
    should be grouped into "root"                             
  File "/root/bin/config-version.sh"
    should exist               
    should be mode 750                                       
    should be owned by "root"
    should be grouped into "root"
  Command "/root/bin/config-version.sh"
    exit_status                                                 
      should eq 0
  Command "/root/bin/config-version.sh -c 9999"
    exit_status
      should eq 1
  File "/etc/resolv.conf"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/network"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/wireless"
    should be symlink
    should be owned by "root"
    should be grouped into "root"

Finished in 11.46 seconds (files took 0.6406 seconds to load)
30 examples, 0 failures
```

@davidelang @owendelong @kylerisse I think all of the concerns have been address. Ill merge this in the morning unless someone has any objections

**Comment by kylerisse** [comment]
Looks good @sarcasticadmin . Github wont let me approve since I opened this one, but I approve and appreciate all the effort to get this past the finish line. 🚢

**Pull request merged**